modules/gapi/src/backends/cpu/gnnparsers.cpp

```diff
@@ -246,6 +246,33 @@ void parseSSD(const cv::Mat&  in_ssd_result,
     }
 }
 
+static void checkYoloDims(const MatSize& d) {
+    // Accept 1x1x1xN, 1x1xN, 1xN,
+    // 1x13x13xN and 13x13xN
+    GAPI_Assert(d.dims() >= 1);
+    const auto checkOneDimCase = [&](){
+        GAPI_Assert(d[d.dims()-1]%(5*13*13) == 0);
+        for (int i = 0; i < d.dims()-1; i++) {
+            GAPI_Assert(d[i] == 1);
+        }
+    };
+
+    if (d.dims() >= 3) {
+        if (d[d.dims()-2] == 13) {
+            GAPI_Assert(d[d.dims()-1]%5 == 0);
+            GAPI_Assert(d[d.dims()-2] == 13);
+            GAPI_Assert(d[d.dims()-3] == 13);
+            for (int i = 0; i < d.dims()-3; i++) {
+                GAPI_Assert(d[i] == 1);
+            }
+        } else {
+            checkOneDimCase();
```

Comment:
You can avoid lambda (looks bad in assertion messages) and expand its content below if you put `return` after `for`.

Also you may want to cache `d.dims()`: `const int d = d.dims()`

Reply:
Fixed, thanks!